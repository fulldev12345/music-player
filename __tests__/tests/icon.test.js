import { mount } from 'enzyme'
import React from 'react'
import lyric from '../../example/lyric'
import ReactJkMusicPlayer from '../../src'

const PauseIcon = () => <div>1</div>
const PlayIcon = () => <div>1</div>
const DestroyIcon = () => <div>1</div>
const CloseIcon = () => <div>1</div>
const DeleteIcon = () => <div>1</div>
const DownloadIcon = () => <div>1</div>
const ToggleIcon = () => <div>1</div>
const LyricIcon = () => <div>1</div>
const VolumeIcon = () => <div>1</div>
const MuteIcon = () => <div>1</div>
const NextIcon = () => <div>1</div>
const PrevIcon = () => <div>1</div>
const PlayListsIcon = () => <div>1</div>
const ReloadIcon = () => <div>1</div>
const LoopIcon = () => <div>1</div>
const OrderIcon = () => <div>1</div>
const OrderLoopIcon = () => <div>1</div>
const ShuffleIcon = () => <div>1</div>
const LoadingIcon = () => <div>1</div>
const PackUpPanelMobileIcon = () => <div>1</div>
const EmptyIcon = () => <div>1</div>

const createPlayer = (props) => (
  <ReactJkMusicPlayer
    mode="full"
    audioLists={[
      {
        musicSrc: 'xx',
        name: 'audioName',
        lyric,
      },
      {
        musicSrc: 'xx2',
        name: 'audioName2',
      },
    ]}
    icon={{
      pause: <PauseIcon />,
      play: <PlayIcon />,
      destroy: <DestroyIcon />,
      close: <CloseIcon />,
      delete: <DeleteIcon />,
      download: <DownloadIcon />,
      toggle: <ToggleIcon />,
      lyric: <LyricIcon />,
      volume: <VolumeIcon />,
      mute: <MuteIcon />,
      next: <NextIcon />,
      prev: <PrevIcon />,
      playLists: <PlayListsIcon />,
      reload: <ReloadIcon />,
      loop: <LoopIcon />,
      order: <OrderIcon />,
      orderLoop: <OrderLoopIcon />,
      shuffle: <ShuffleIcon />,
      loading: <LoadingIcon />,
      packUpPanelMobile: <PackUpPanelMobileIcon />,
      empty: <EmptyIcon />,
    }}
    {...props}
  />
)

describe('Player custom icon test', () => {
  it('should render custom icon', () => {
    const wrapper = mount(createPlayer())
    wrapper.find('.audio-lists-btn').simulate('click')
    expect(wrapper.render()).toMatchSnapshot()
  })

  it('should render custom icon in mobile mode', () => {
    const wrapper = mount(createPlayer())
    wrapper.setState({ isMobile: true })
    expect(wrapper.render()).toMatchSnapshot()
  })
})
