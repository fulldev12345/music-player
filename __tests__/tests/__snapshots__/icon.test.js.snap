// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player custom icon test should render custom icon 1`] = `
<div
  class="react-jinke-music-player-main dark-theme"
  tabindex="-1"
>
  <div
    class="music-player-panel translate"
  >
    <section
      class="panel-content"
    >
      <div
        class="img-content img-rotate img-rotate-pause"
        style="background-image: url();"
      />
      <div
        class="progress-bar-content"
      >
        <span
          class="audio-title"
          title="audioName"
        >
          audioName
        </span>
        <section
          class="audio-main"
        >
          <span
            class="current-time"
            title="00:00"
          >
            00:00
          </span>
          <div
            class="progress-bar"
          >
            <div
              class="progress-load-bar"
              style="width: 0%;"
            />
            <div
              class="rc-slider"
            >
              <div
                class="rc-slider-rail"
              />
              <div
                class="rc-slider-track"
                style="left: 0%;"
              />
              <div
                class="rc-slider-step"
              />
              <div
                aria-disabled="false"
                aria-valuemax="0"
                aria-valuemin="0"
                aria-valuenow="0"
                class="rc-slider-handle"
                role="slider"
                style="transform: translateX(-50%);"
                tabindex="0"
              />
              <div
                class="rc-slider-mark"
              />
            </div>
          </div>
          <span
            class="duration"
            title="00:00"
          >
            00:00
          </span>
        </section>
      </div>
      <div
        class="player-content"
      >
        <span
          class="group"
        >
          <span
            class="group prev-audio"
            title="Previous track"
          >
            <div>
              1
            </div>
          </span>
          <span
            class="group play-btn"
            title="Click to play"
          >
            <div>
              1
            </div>
          </span>
          <span
            class="group next-audio"
            title="Next track"
          >
            <div>
              1
            </div>
          </span>
        </span>
        <span
          class="group reload-btn"
          title="Reload"
        >
          <div>
            1
          </div>
        </span>
        <span
          class="group audio-download"
          title="Download"
        >
          <div>
            1
          </div>
        </span>
        <span
          class="group theme-switch"
        >
          <button
            aria-checked="false"
            class="rc-switch theme-switch-container"
            role="switch"
            title="Dark/Light mode"
            type="button"
          >
            <span
              class="rc-switch-inner"
            >
              D
            </span>
          </button>
        </span>
        <span
          class="group play-sounds"
          title="Volume"
        >
          <span
            class="sounds-icon"
          >
            <div>
              1
            </div>
          </span>
          <div
            class="rc-slider sound-operation"
          >
            <div
              class="rc-slider-rail"
            />
            <div
              class="rc-slider-track"
              style="left: 0%; width: 100%;"
            />
            <div
              class="rc-slider-step"
            />
            <div
              aria-disabled="false"
              aria-valuemax="1"
              aria-valuemin="0"
              aria-valuenow="1"
              class="rc-slider-handle"
              role="slider"
              style="left: 100%; transform: translateX(-50%);"
              tabindex="0"
            />
            <div
              class="rc-slider-mark"
            />
          </div>
        </span>
        <span
          class="group loop-btn"
          title="Play in order"
        >
          <div>
            1
          </div>
        </span>
        <span
          class="group audio-lists-btn"
          title="Playlists"
        >
          <span
            class="audio-lists-icon"
          >
            <div>
              1
            </div>
          </span>
          <span
            class="audio-lists-num"
          >
            2
          </span>
        </span>
        <span
          class="group hide-panel"
          title="Minimize"
        >
          <div>
            1
          </div>
        </span>
      </div>
    </section>
  </div>
  <div
    class="audio-lists-panel show"
  >
    <div
      class="audio-lists-panel-header"
    >
      <h2
        class="audio-lists-panel-header-title"
      >
        <span>
          Playlists / 
        </span>
        <span
          class="audio-lists-panel-header-num"
        >
          2
        </span>
        <span
          class="audio-lists-panel-header-actions"
        >
          <span
            class="audio-lists-panel-header-delete-btn"
            title="Delete audio lists"
          >
            <div>
              1
            </div>
          </span>
          <span
            class="audio-lists-panel-header-line"
          />
          <span
            class="audio-lists-panel-header-close-btn"
            title="Close"
          >
            <div>
              1
            </div>
          </span>
        </span>
      </h2>
    </div>
    <div
      class="audio-lists-panel-content"
    >
      <ul
        class="audio-lists-panel-content-wrap"
      >
        <li
          class="audio-item playing pause"
          title="Click to play"
        >
          <span
            class="group player-status"
          >
            <span
              class="player-icons"
            >
              <div>
                1
              </div>
            </span>
          </span>
          <span
            class="group player-name"
            title="audioName"
          >
            audioName
          </span>
          <span
            class="group player-singer"
          />
          <span
            class="group player-delete"
            title="Click to delete audioName"
          >
            <div>
              1
            </div>
          </span>
        </li>
        <li
          class="audio-item pause"
          title="Click to play"
        >
          <span
            class="group player-status"
          >
            <span
              class="player-icons"
            />
          </span>
          <span
            class="group player-name"
            title="audioName2"
          >
            audioName2
          </span>
          <span
            class="group player-singer"
          />
          <span
            class="group player-delete"
            title="Click to delete audioName2"
          >
            <div>
              1
            </div>
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="play-mode-title"
  >
    Play in order
  </div>
  <audio
    class="music-player-audio"
    src="xx"
    title="audioName"
  />
</div>
`;

exports[`Player custom icon test should render custom icon in mobile mode 1`] = `
<div
  class="react-jinke-music-player-main dark-theme"
  tabindex="-1"
>
  <div
    class="react-jinke-music-player-mobile default-bg"
  >
    <div
      class="react-jinke-music-player-mobile-play-model-tip"
    >
      <span
        class="react-jinke-music-player-mobile-play-model-tip-title"
      >
        <span
          class="group loop-btn"
          title="Play in order"
        >
          <div>
            1
          </div>
        </span>
      </span>
      <span
        class="react-jinke-music-player-mobile-play-model-tip-text"
      >
        Play in order
      </span>
    </div>
    <div
      class="react-jinke-music-player-mobile-header group"
    >
      <div
        class="react-jinke-music-player-mobile-header-title"
        title="audioName"
      >
        audioName
      </div>
      <div
        class="react-jinke-music-player-mobile-header-right"
      >
        <div>
          1
        </div>
      </div>
    </div>
    <div
      class="react-jinke-music-player-mobile-singer text-center group"
    >
      <span
        class="react-jinke-music-player-mobile-singer-name"
        title=""
      />
    </div>
    <div
      class="react-jinke-music-player-mobile-switch text-center group"
    >
      <span
        class="group theme-switch"
      >
        <button
          aria-checked="false"
          class="rc-switch theme-switch-container"
          role="switch"
          title="Dark/Light mode"
          type="button"
        >
          <span
            class="rc-switch-inner"
          >
            D
          </span>
        </button>
      </span>
    </div>
    <div
      class="react-jinke-music-player-mobile-cover text-center"
    >
      <img
        alt="cover"
        class="cover img-rotate-pause"
        src=""
      />
    </div>
    <div
      class="react-jinke-music-player-mobile-progress group"
    >
      <span
        class="current-time"
      >
        00:00
      </span>
      <span
        class="react-jinke-music-player-mobile-progress-bar"
      >
        <div
          class="progress-load-bar"
          style="width: 0%;"
        />
        <div
          class="rc-slider"
        >
          <div
            class="rc-slider-rail"
          />
          <div
            class="rc-slider-track"
            style="left: 0%;"
          />
          <div
            class="rc-slider-step"
          />
          <div
            aria-disabled="false"
            aria-valuemax="0"
            aria-valuemin="0"
            aria-valuenow="0"
            class="rc-slider-handle"
            role="slider"
            style="transform: translateX(-50%);"
            tabindex="0"
          />
          <div
            class="rc-slider-mark"
          />
        </div>
      </span>
      <span
        class="duration text-right"
      >
        00:00
      </span>
    </div>
    <div
      class="react-jinke-music-player-mobile-toggle text-center group"
    >
      <span
        class="group prev-audio"
        title="Previous track"
      >
        <div>
          1
        </div>
      </span>
      <span
        class="group play-btn"
        title="Click to play"
      >
        <div>
          1
        </div>
      </span>
      <span
        class="group next-audio"
        title="Next track"
      >
        <div>
          1
        </div>
      </span>
    </div>
    <div
      class="react-jinke-music-player-mobile-operation group"
    >
      <ul
        class="items"
      >
        <li
          class="item"
        >
          <span
            class="group loop-btn"
            title="Play in order"
          >
            <div>
              1
            </div>
          </span>
        </li>
        <li
          class="item"
        >
          <span
            class="group audio-download"
            title="Download"
          >
            <div>
              1
            </div>
          </span>
        </li>
        <li
          class="item"
        >
          <span
            class="group reload-btn"
            title="Reload"
          >
            <div>
              1
            </div>
          </span>
        </li>
        <li
          class="item"
        >
          <div>
            1
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="audio-lists-panel audio-lists-panel-mobile"
  >
    <div
      class="audio-lists-panel-header"
    >
      <h2
        class="audio-lists-panel-header-title"
      >
        <span>
          Playlists / 
        </span>
        <span
          class="audio-lists-panel-header-num"
        >
          2
        </span>
        <span
          class="audio-lists-panel-header-actions"
        >
          <span
            class="audio-lists-panel-header-delete-btn"
            title="Delete audio lists"
          >
            <div>
              1
            </div>
          </span>
          <span
            class="audio-lists-panel-header-line"
          />
          <span
            class="audio-lists-panel-header-close-btn"
            title="Close"
          >
            <div>
              1
            </div>
          </span>
        </span>
      </h2>
    </div>
    <div
      class="audio-lists-panel-content"
    >
      <ul
        class="audio-lists-panel-content-wrap"
      >
        <li
          class="audio-item playing pause"
          title="Click to play"
        >
          <span
            class="group player-status"
          >
            <span
              class="player-icons"
            >
              <div>
                1
              </div>
            </span>
          </span>
          <span
            class="group player-name"
            title="audioName"
          >
            audioName
          </span>
          <span
            class="group player-singer"
          />
          <span
            class="group player-delete"
            title="Click to delete audioName"
          >
            <div>
              1
            </div>
          </span>
        </li>
        <li
          class="audio-item pause"
          title="Click to play"
        >
          <span
            class="group player-status"
          >
            <span
              class="player-icons"
            />
          </span>
          <span
            class="group player-name"
            title="audioName2"
          >
            audioName2
          </span>
          <span
            class="group player-singer"
          />
          <span
            class="group player-delete"
            title="Click to delete audioName2"
          >
            <div>
              1
            </div>
          </span>
        </li>
      </ul>
    </div>
  </div>
  <audio
    class="music-player-audio"
    src="xx"
    title="audioName"
  />
</div>
`;
